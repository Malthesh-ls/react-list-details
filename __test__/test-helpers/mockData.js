const mockRespUniversityData = [
  {
    "state-province": "Abu Dhabi",
    "domains": [
      "mbzuai.ac.ae"
    ],
    "web_pages": [
      "https://mbzuai.ac.ae/"
    ],
    "name": "Mohamed bin Zayed University of Artificial Intelligence (MBZUAI)",
    "alpha_two_code": "AE",
    "country": "United Arab Emirates"
  },
  {
    "domains": [
      "acd.ac.ae"
    ],
    "country": "United Arab Emirates",
    "state-province": null,
    "alpha_two_code": "AE",
    "web_pages": [
      "http://www.acd.ac.ae/"
    ],
    "name": "American College Of Dubai"
  },
];

const mockUniversityData = [
  {
    "state_province": "Abu Dhabi",
    "domains": [
      "mbzuai.ac.ae"
    ],
    "web_pages": [
      "https://mbzuai.ac.ae/"
    ],
    "name": "Mohamed bin Zayed University of Artificial Intelligence (MBZUAI)",
    "alpha_two_code": "AE",
    "country": "United Arab Emirates",
    "uid": 1
  },
  {
    "domains": [
      "acd.ac.ae"
    ],
    "country": "United Arab Emirates",
    "state_province": null,
    "alpha_two_code": "AE",
    "web_pages": [
      "http://www.acd.ac.ae/"
    ],
    "name": "American College Of Dubai",
    "uid": 3
  },
];

const mockColumns = [
  { "label": "Name", "accessor": "name", "sortable": true, "sortbyOrder": "asc" },
  { "label": "Alpha Code", "accessor": "alpha_two_code", "sortable": true, },
  { "label": "State Province", "accessor": "state_province", "sortable": true },
  { "label": "Country", "accessor": "country", "sortable": true },
  { "label": "Action", "accessor": "action", "sortable": false },
];

export {
  mockRespUniversityData,
  mockUniversityData,
  mockColumns,
}