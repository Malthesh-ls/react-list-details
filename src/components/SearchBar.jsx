import { SerachIcon } from '../components/common/Icon';

const SearchBar = ({ data, setSearchData }) => {
  const handleSubmit = (e) => e.preventDefault();
  const handleSearchOnChange = (e) => {
    if (!e.target.value) return setSearchData(data);

    const resultsArry = data.filter(item => item.name.toLowerCase().includes(e.target.value.toLowerCase()));
    setSearchData(resultsArry);
  }
  return (
    <div className='searchBar'>
      <form className='search' onClick={handleSubmit}>
        <input
          className='searchInput'
          id='search'
          type='text'
          onChange={handleSearchOnChange}
        />
        <button className='searchButton'><SerachIcon /></button>
      </form>
    </div>
  )
}

export default SearchBar