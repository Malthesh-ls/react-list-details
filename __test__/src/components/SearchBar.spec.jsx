import renderer from 'react-test-renderer';
import SearchBar from '../../../src/components/SearchBar';
import { mockUniversityData } from '../../test-helpers/mockData';

describe('SearchBar Tests', () => {
  const props = {
    data: mockUniversityData,
    setData: jest.fn(),
  }
  it('should match snapshot', () => {
    const snapshot = renderer
      .create(<SearchBar {...props} />)
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});