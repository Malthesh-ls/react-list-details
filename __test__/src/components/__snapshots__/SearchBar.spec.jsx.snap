// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Tests should match snapshot 1`] = `
<div
  className="searchBar"
>
  <form
    className="search"
    onClick={[Function]}
  >
    <input
      className="searchInput"
      id="search"
      onChange={[Function]}
      type="text"
    />
    <button
      className="searchButton"
    >
      <svg
        fill="currentColor"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.742 10.344a6.5 6.5 0 1 0-1.397 1.398h-.001q.044.06.098.115l3.85 3.85a1 1 0 0 0 1.415-1.414l-3.85-3.85a1 1 0 0 0-.115-.1zM12 6.5a5.5 5.5 0 1 1-11 0 5.5 5.5 0 0 1 11 0"
        />
      </svg>
    </button>
  </form>
</div>
`;
